meta_test.go

```diff
@@ -357,6 +358,43 @@ func (s *MetaSuite) TestInvalidSeries(c *gc.C) {
 	}
 }
 
+// TestMinJujuVersion ensures that valid MinJujuVersion values are parsed
+// correctly when specified in the charm metadata.
+func (s *MetaSuite) TestMinJujuVersion(c *gc.C) {
+	// series not specified
+	meta, err := charm.ReadMeta(strings.NewReader(dummyMetadata))
+	c.Assert(err, gc.IsNil)
+	c.Check(meta.Series, gc.HasLen, 0)
+	charmMeta := fmt.Sprintf("%s\nmin-juju-version: ", dummyMetadata)
+	vals := []version.Number{
+		{Major: 1, Minor: 25},
+		{Major: 1, Minor: 25, Tag: "alpha1"},
+		{Major: 1, Minor: 25, Patch: 1},
+	}
+	for _, ver := range vals {
+		val := charmMeta + ver.String()
+		meta, err = charm.ReadMeta(strings.NewReader(val))
+		c.Assert(err, gc.IsNil)
+		c.Assert(meta.MinJujuVersion, gc.Equals, ver)
+	}
+}
+
+// TestInvalidSeries ensures that invalid series values cause a parse error
+// when specified in the charm metadata.
+func (s *MetaSuite) TestInvalidMinJujuVersion(c *gc.C) {
+	_, err := charm.ReadMeta(strings.NewReader(dummyMetadata + "\nmin-juju-version: invalid-version"))
+
+	c.Check(err, gc.ErrorMatches, `invalid version "invalid-version"`)
```

Comment:
can the error message be 'invalid min-version "foo"' so the user knows what attribute is wrong